### PR TITLE
Remove mkinitrd script.

### PR DESCRIPTION
**Pull request by cwize1**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [ ] Ready to merge

---

###### Summary 

This changes removes the mkinitrd script and replaces all calls with direct calls to dracut.

In addition, the /boot/initrd.img-<kver> file has been renamed to /boot/initramfs-<kver>.img to match the default name used by dracut. This avoids needing to add a patch to dracut to change the default name of the initrd file.

Also, previously the mkinitrd script passed the `--hostonly` flag to dracut to ensure the hostonly option was enabled. This has been replaced with setting the hostonly only in the `/etc/dracut.conf` file instead. One nice advantage this provides is that it allows a user to override this default by adding a config file under the `/etc/dracut.conf.d` directory.

###### Change Log  

- Remove mkinitrd script.
- Rename `/boot/initrd.img-<kver>` to `/boot/initramfs-<kver>.img`.
- Set `hostonly` to `yes` in `/etc/dracut.conf`.
- Call dracut directly in iniramfs package post-install scripts.

###### Does this affect the toolchain?  

YES, but only a kernel-headers version bump.

###### Test Methodology

- Built image locally and booted it in a VM.

### CODE REVIEW

File: SPECS-SIGNED/kernel-signed/kernel-signed.spec [hunk 1]
```diff
@@ -147,6 +147,9 @@ echo "initrd of kernel %{uname_r} removed" >&2
 %exclude /module_info.ld
 
 %changelog
+* Fri Feb 23 2024 Chris Gunn <chrisgun@microsoft.com> - 5.15.135.1-3
```
Comment (cwize1) [inline SPECS-SIGNED/kernel-signed/kernel-signed.spec:150]: Fix version string in changelog.

**Review by PawelWMS — APPROVED** [review]
The toolkit changes look good.

**Review by ddstreetmicrosoft — CHANGES_REQUESTED** [review]
Several things, most importantly we definitely need to revisit the (incomplete) `kexec-tools` upgrade, and we absolutely need to make sure `kdump` is not using the system initrd.

I'd also like to understand why we have the `initramfs` package at all.

Finally, that `kernel-rt` grub config stuff almost certainly is no longer needed and should be removed.

File: SPECS/initramfs/initramfs.spec [hunk 2]
```diff
@@ -130,6 +114,10 @@ echo "initramfs" %{version}-%{release} "postun" >&2
 %dir %{_localstatedir}/lib/initramfs/kernel
 
 %changelog
+* Fri Feb 23 2024 Chris Gunn <chrisgun@microsoft.com> - 3.0-2
+- Call dracut instead of mkinitrd
+- Rename initrd.img-<kver> to initramfs-<kver>.img
+
```
[the remark this answers is not in the document]
Comment (cwize1) [reply SPECS/initramfs/initramfs.spec:120]: The purpose of the initramfs package is to update the initramfs file when a package is installed that affects the initramfs config.

For example, the dracut-megaraid package adds a config file under /etc/dracut.conf.d that adds the megaraid module to the initramfs. After the dracut-megaraid package is installed, the post-install scripts in the initramfs package will trigger and the initramfs file will be updated.

File: SPECS/dracut/dracut.spec [hunk 3]
```diff
@@ -27,12 +26,16 @@ Patch:          fix-functions-Avoid-calling-grep-with-PCRE-P.patch
...
 Patch:          0002-disable-xattr.patch
-Patch:          0003-fix-initrd-naming-for-photon.patch
 Patch:          0006-dracut.sh-validate-instmods-calls.patch
 Patch:          0007-feat-dracut.sh-support-multiple-config-dirs.patch
 Patch:          0008-fix-dracut-systemd-rootfs-generator-cannot-write-out.patch
 Patch:          0009-install-systemd-executor.patch
 
+# kdump currently uses the host system's initrd when enrolling a crash kernel
+# and initrd. There is a limitation where the kdump initrd must be generated
+# with dracut in hostonly mode. So, set hostonly as the default.
```
[the remark this answers is not in the document]
Comment (cwize1) [reply SPECS/dracut/dracut.spec:36]: I’m just maintaining existing behavior here. If we want to change the default, we should do it in a different PR.

File: SPECS/dracut/dracut-conf-add-defaults.patch [hunk 4]
```diff
@@ -0,0 +1,10 @@
+diff --git a/dracut.conf b/dracut.conf
+index d6317a0f..390c377e 100644
+--- a/dracut.conf
++++ b/dracut.conf
+@@ -1,3 +1,5 @@
+ # PUT YOUR CONFIG IN separate files
+ # in /etc/dracut.conf.d named "<name>.conf"
+ # SEE man dracut.conf(5) for options
++
++hostonly=yes
```
[the remark this answers is not in the document]
Comment (cwize1) [reply SPECS/dracut/dracut-conf-add-defaults.patch:10]: I believe the purpose of this file is to store the default values for distro. This allows the user to add files under /etc/dracut.conf.d/ to override the distro defaults, without needing to worry about file ordering.

[the remark this answers is not in the document]
Comment (cwize1) [reply SPECS/dracut/dracut-conf-add-defaults.patch:10]: Explanation for this as a default value is in the spec file. This default was originally enforced by the initramfs package and couldn’t be overridden by the user.

File: SPECS-EXTENDED/kernel-rt/kernel-rt.spec [hunk 5]
```diff
@@ -310,7 +310,7 @@ echo "initrd generation of kernel %{uname_r} will be triggered later" >&2
 
 %triggerun -- initramfs
```
[the remark this answers is not in the document]
Comment (cwize1) [reply SPECS-EXTENDED/kernel-rt/kernel-rt.spec:311]: The Fedora kernel package also uses the dracut command in post-install scripts. So, I don’t think using kernel-install removes the need for dracut post-install scripts. If anything, this change actually brings us closer to what Fedora is doing.

File: SPECS-EXTENDED/kernel-rt/kernel-rt.spec [hunk 6]
```diff
@@ -249,7 +249,7 @@ cat > %{buildroot}/boot/linux-%{uname_r}.cfg << "EOF"
 # GRUB Environment Block
```
[the remark this answers is not in the document]
Comment (cwize1) [reply SPECS-EXTENDED/kernel-rt/kernel-rt.spec:249]: This stuff was only removed from the kernel package. The alternate kernel packages weren’t updated.

IIRC, there are plans to remove all the alternate kernels. So, that might be why these packages weren’t updated.

File: toolkit/tools/imagegen/installutils/installutils.go [hunk 7]
```diff
@@ -778,11 +778,12 @@ func updateInitramfsForEncrypt(installChroot *safechroot.Chroot) (err error) {
 		const (
 			libModDir     = "/lib/modules"
 			dracutModules = "dm crypt crypt-gpg crypt-loop lvm"
-			initrdPrefix  = "/boot/initrd.img-"
+			initrdPrefix  = "/boot/initramfs-"
+			initrdSuffix  = ".img"
```
[the remark this answers is not in the document]
Comment (cwize1) [reply toolkit/tools/imagegen/installutils/installutils.go:782]: This is just updating existing code.

File: SPECS/kexec-tools/kdumpctl [hunk 8]
```diff
@@ -311,16 +311,16 @@ setup_initrd()
 
 	kdump_kernel="${KDUMP_BOOTDIR}/${KDUMP_IMG}-${kdump_kver}${KDUMP_IMG_EXT}"
 
-	DEFAULT_INITRD="${KDUMP_BOOTDIR}/initrd.img-`uname -r`"
-	DEFAULT_INITRD_BAK="${KDUMP_BOOTDIR}/.initrd.img-`uname -r`.default"
+	DEFAULT_INITRD="${KDUMP_BOOTDIR}/initramfs-`uname -r`".img
+	DEFAULT_INITRD_BAK="${KDUMP_BOOTDIR}/.initramfs-`uname -r`.default".img
```
[the remark this answers is not in the document]
Comment (cwize1) [reply SPECS/kexec-tools/kdumpctl:315]: Does this have any impact on this change? Or is it just something we should deal with as a separate task?

File: SPECS/dracut/dracut-conf-add-defaults.patch (hunk 4 above)
[the remark this answers is not in the document]
Comment (ddstreetmicrosoft) [reply SPECS/dracut/dracut-conf-add-defaults.patch:10]: this absolutely should not be in here, but i can remove it in a later pr

File: SPECS/dracut/dracut.spec (hunk 3 above)
[the remark this answers is not in the document]
Comment (ddstreetmicrosoft) [reply SPECS/dracut/dracut.spec:36]: yeah we can fix things in a different pr

File: SPECS/initramfs/initramfs.spec (hunk 2 above)
[the remark this answers is not in the document]
Comment (ddstreetmicrosoft) [reply SPECS/initramfs/initramfs.spec:120]: well my point was simple triggers don't actually need an entire package, but this is another thing that we can fix in a separate pr

**Review by ddstreetmicrosoft — APPROVED** [review]
definitely multiple issues with our overall setup but as you pointed out you're not trying to fix everything in this pr, just make a minor tweak, and that tweak looks fine (mostly), and i'll try to keep track of the existing issues and fix in separate prs

**Review by dmcilvaney — APPROVED** [review]

**Comment by LaszloGombos** [comment]
A message from dracut upstream regarding the intention of standardizing the naming of the generated initramfs.

https://github.com/dracut-ng/dracut-ng/pull/160
https://github.com/uapi-group/specifications/issues/100

You should perhaps consider keeping the initrd-$KVER naming schema.

**Pull request merged**